data/sql_updates/totals_pac.sql

```diff
@@ -4,84 +4,106 @@ create materialized view ofec_totals_pacs_mv_tmp as
-    sum(pnp.loan_repymts_made_per) as loan_repayments_made,
-    sum(pnp.loan_repymts_received_per) as loan_repayments_received,
-    sum(pnp.loans_made_per) as loans_made,
-    sum(pnp.net_contb_per) as net_contributions,
-    sum(pnp.net_op_exp_per) as net_operating_expenditures,
-    sum(pnp.non_alloc_fed_elect_actvy_per) as non_allocated_fed_election_activity,
-    sum(pnp.ttl_nonfed_tranf_per) as nonfed_transfers,
-    sum(pnp.ttl_op_exp_per) as offsets_to_operating_expenditures,
-    sum(pnp.ttl_op_exp_per) as operating_expenditures,
-    sum(pnp.other_disb_per) as other_disbursements,
-    sum(pnp.other_fed_op_exp_per) as other_fed_operating_expenditures,
-    sum(pnp.other_fed_receipts_per) as other_fed_receipts,
-    sum(pnp.other_pol_cmte_contb_per_i) as other_political_committee_contributions,
-    sum(pnp.other_pol_cmte_refund) as refunded_other_political_committee_contributions,
-    sum(pnp.pol_pty_cmte_contb_per_i) as political_party_committee_contributions,
-    sum(pnp.pol_pty_cmte_refund) as refunded_political_party_committee_contributions,
-    sum(pnp.ttl_receipts) as receipts,
-    sum(pnp.shared_fed_actvy_fed_shr_per) as shared_fed_activity,
-    sum(pnp.shared_fed_actvy_nonfed_per) as shared_fed_activity_nonfed,
-    sum(pnp.shared_fed_op_exp_per) as shared_fed_operating_expenditures,
-    sum(pnp.shared_nonfed_op_exp_per) as shared_nonfed_operating_expenditures,
-    sum(pnp.tranf_from_affiliated_pty_per) as transfers_from_affiliated_party,
-    sum(pnp.tranf_from_nonfed_acct_per) as transfers_from_nonfed_account,
-    sum(pnp.tranf_from_nonfed_levin_per) as transfers_from_nonfed_levin,
-    sum(pnp.tranf_to_affliliated_cmte_per) as transfers_to_affiliated_committee,
-    max(last.rpt_tp_desc) as last_report_type_full,
-    max(last.begin_image_num) as last_beginning_image_number,
-    max(last.coh_cop) as last_cash_on_hand_end_period,
-    max(last.debts_owed_by_cmte) as last_debts_owed_by_committee,
-    max(last.rpt_yr) as last_report_year
-from
-    fec_vsum_f3x_vw pnp
-    inner join ofec_committee_detail_mv_tmp comm_dets on cmte_id = comm_dets.committee_id
-    left join last using (cmte_id, election_cycle)
-where
-    pnp.most_recent_filing_flag like 'Y'
-    and election_cycle >= :START_YEAR
-    --yuk, there's got to be a better way!--
-    and (comm_dets.committee_type like 'N' or comm_dets.committee_type like 'Q'
-    or comm_dets.committee_type like 'O' or comm_dets.committee_type like 'V'
-    or comm_dets.committee_type like 'W')
-group by
-    cmte_id,
-    comm_dets.name,
-    comm_dets.committee_type,
-    pnp.election_cycle
+), cash_beginning_period as (
+	  select distinct on (pac.cmte_id, pac.election_cycle)
+	      pac.cmte_id                           as committee_id,
+	      pac.election_cycle                    as cycle,
+	      pac.coh_bop as cash_on_hand
+	  from
+        fec_vsum_f3x_vw pac
+        inner join ofec_committee_detail_mv_tmp comm_dets on pac.cmte_id = comm_dets.committee_id
```

Comment:
I am not sure why there is a join here. It seems like we just need the earliest cash on hand and we can do that by ordering by coverage end date or coverage start date. Maybe something along the lines of: ```
cash_beginning_period as (
   select distinct on (cmte_id, election_cycle)
       cmte_id as committee_id,
       election_cycle as cycle,
       coh_bop as cash_on_hand,
       cvg_end_dt
  from fec_vsum_f3x_vw 
  order by  cvg_end_dt  acc ```